### PR TITLE
Fix java invalid base dir

### PR DESCRIPTION
**Pull request by abrooksv**
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)

## Description
Query project root from Maven/Gradle to determine the base directory for Java

When using Java handler based build, query Maven/Gradle for the project root.

## Related Issue(s)
#868, #857

## Testing
We now actually import the projects and then try to build them

## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by abrooksv** [comment]
Opps, need to rebase on to clean origin/master

File: jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/java/JavaTestUtils.kt [hunk 1]
```diff
@@ -50,6 +75,68 @@ internal fun HeavyJavaCodeInsightTestFixtureRule.setUpGradleProject(): PsiClass
...
+            error.set(errorMessage)
+        }
+    }
+
+    val importSpecBuilder = ImportSpecBuilder(project, GradleConstants.SYSTEM_ID)
+        .callback(refreshCallback)
+        .forceWhenUptodate()
+        .use(ProgressExecutionMode.MODAL_SYNC)
+
+    ExternalSystemUtil.refreshProjects(importSpecBuilder)
```
Comment (kiiadi) [inline jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/java/JavaTestUtils.kt]: does this happen synchronously?

Comment (abrooksv) [reply jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/java/JavaTestUtils.kt]: Yes, that is controlled by the ProgressExecutionMode enum 2 lines up

File: jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilderTest.kt [hunk 2]
```diff
@@ -130,4 +131,24 @@ class JavaLambdaBuilderTest : BaseLambdaBuilderTest() {
...
+                    return input.toUpperCase();
+                }
+            }
+            """.trimIndent()
+        )
+
+        assertThatThrownBy {
+            buildLambda(projectRule.module, handlerPsi, Runtime.JAVA8, "com.example.SomeClass")
+        }.isInstanceOf(IllegalStateException::class.java)
+            .hasMessageEndingWith("is not managed by Maven or Gradle")
```
Comment (zhangzhx) [inline jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilderTest.kt]: Hmm, as the exception message is managed by the localization properties, I'm wondering if it is appropriate to test the exception message

Comment (abrooksv) [reply jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilderTest.kt]: Ya it is fragile, but it does enforce the correct exception is thrown and not a random different IllegalState

Comment (zhangzhx) [reply jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilderTest.kt]: I understand you want to enforce the correct exception. It's better to fix the fragile test, but I'm fine with it too considering we are not expanding the localization in a short term.

Comment (abrooksv) [reply jetbrains-core/it/software/aws/toolkits/jetbrains/services/lambda/java/JavaLambdaBuilderTest.kt]: I can just use the message() call here as well. I'll make the change tomorrow

**Pull request merged**